homework/index.js

```diff
@@ -30,15 +32,138 @@
     return elem;
   }
 
-  function main(url) {
-    fetchJSON(url, (err, data) => {
-      const root = document.getElementById('root');
-      if (err) {
-        createAndAppend('div', root, { text: err.message, class: 'alert-error' });
+  // Creating the repository-details part
+  function createHyfRepoDetails(leftDiv, hyfRepo) {
+    const table = createAndAppend('table', leftDiv, { class: 'table' });
+    const tBody = createAndAppend('tbody', table);
+    const details = ['Repository', 'Description', 'Forks', 'Updated'];
+    details.forEach(detail => {
+      const tr = createAndAppend('tr', tBody, { class: `${detail.toLowerCase()}-row` });
+      createAndAppend('td', tr, {
+        text: `${detail}: `,
+        class: `label ${detail.toLowerCase()}-head`,
+      });
+      createAndAppend('td', tr, { class: `${detail.toLowerCase()}-data` });
+    });
+    const repoName = document.querySelector('.repository-data');
+    createAndAppend('a', repoName, {
+      href: hyfRepo.html_url,
+      target: '_blank',
+      text: hyfRepo.name,
+      class: 'repo-name',
+    });
+    if (hyfRepo.description === null) {
+      document.querySelector('.description-row').setAttribute('class', 'hide-description-row');
+    } else {
+      document.querySelector('.description-data').innerText = hyfRepo.description;
+    }
+    document.querySelector('.forks-data').innerText = hyfRepo.forks;
+    document.querySelector('.updated-data').innerText = new Date(
+      hyfRepo.updated_at,
+    ).toLocaleString();
+  }
+
+  // Creating the contributions part
+  async function createContributors(rightDiv, url) {
+    createAndAppend('h3', rightDiv, {
+      text: 'Contributions: ',
+      class: 'contributor-header',
+    });
+    try {
+      const contributors = await fetchJSON(url);
+      if (!(contributors && contributors.length)) {
+        createAndAppend('div', rightDiv, {
+          text: 'No contributions found.',
+          class: 'alert-error',
+        });
       } else {
-        createAndAppend('pre', root, { text: JSON.stringify(data, null, 2) });
+        const ul = createAndAppend('ul', rightDiv, {
+          class: 'contributor-list',
+        });
+        contributors.forEach(contributor => {
+          const contributorItem = createAndAppend('li', ul, {
+            class: 'contributor-item',
+          });
+          const contributorLink = createAndAppend('a', contributorItem, {
+            href: contributor.html_url,
+            target: '_blank',
+          });
+          const contributorDiv = createAndAppend('div', contributorLink, {
```

Comment:
I would say that this isn't the best way to do it. Because an anchor element(`a`) is an inline element while a `div` element is a block level element. Here, you are trying to add a block level element inside an inline element which isn't semantically correct. You can perhaps think of changing `div` to `span` which are inline elements. 🙂

Reply:
Thanks for this advise. It's done 👍